src/Drivers/Digipay/Digipay.php

```diff
@@ -220,6 +228,111 @@ protected function oauth()
         return $body['access_token'];
     }
 
+    /**
+     * @see https://docs.mydigipay.com/upg.html#_purchase_reverse
+     * @throws PurchaseFailedException
+     */
+    public function reverse()
+    {
+        if (is_null($digipayTicketType = $this->invoice->getDetail('type'))) {
+            throw new PurchaseFailedException('"type" is required for this method.');
+        }
+
+        if (is_null($trackingCode = $this->invoice->getDetail('trackingCode'))) {
+            throw new PurchaseFailedException('"trackingCode" is required for this method.');
+        }
+
+        $data = [
+            'trackingCode' => $trackingCode,
+            'providerId'    => $this->invoice->getTransactionId() ?? $this->invoice->getDetail('providerId') ?? $this->invoice->getUuid(),
+        ];
+
+        $response = $this
+            ->client
+            ->request(
+                'POST',
+                $this->settings->apiPaymentUrl . self::REVERSE_URL,
+                [
+                    RequestOptions::BODY        => json_encode($data),
+                    RequestOptions::QUERY       => ['type' => $digipayTicketType],
+                    RequestOptions::HEADERS     => [
+                        'Content-Type'    => 'application/json;charset=UTF-8',
+                        'Authorization'   => 'Bearer ' . $this->oauthToken,
+                    ],
+                    RequestOptions::HTTP_ERRORS => false,
+                ]
+            );
+
+        $body = json_decode($response->getBody()->getContents(), true);
+
+        if ($response->getStatusCode() != 200 || $body['result']['code'] != 0) {
+            $message = $body['result']['message'] ?? 'خطا در هنگام درخواست برای برگشت وجه رخ داده است.';
+            throw new InvalidPaymentException($message, (int) $response->getStatusCode());
+        }
+
+        return $body;
+    }
+
+    /**
+     * @see https://docs.mydigipay.com/upg.html#_purchase_delivery
+     * @throws PurchaseFailedException
+     */
+    public function deliver()
```

Comment:
as far as I see, nobody uses this function! this function has not been defined by the DriverInterface, so we should not expose it.

if you want to add this feature, we should define a new interface and then implement it.
I propose the current interface can also be renamed from `DriverInterface` to `Purchaser`.

btw, I am not living in Iran, and I was not able the see the Digipay docs.

Reply:
Only this driver has this method. After verifying the transaction, the user can reverse the ticket after 25 minutes. Additionally, after processing an order, it is necessary to call the delivery method for CREDIT and BNPL.

I am uncertain whether other gateways have this action. 🤔

For the PDF version of the Digipay documentation, you can find it [here](https://github.com/tookantech/iranian-services-documents/blob/main/Payment%20Gateways/Digipay/upgDoc(1).pdf).

Reply:
@khanzadimahdi,

Another problem: how can we call these methods? 😁

Perhaps some gateways have other methods, such as:
- Retrieving transaction lists on `IDPay`
- Methods like `bpVerifyRequest`, `bpInquiryRequest`, `bpInquiryRequest`, etc., on `BehpardakhtMelat`
- The `Settlement` method on `AsanPardakht`